app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -59,8 +56,8 @@ fun HomeScreen(
 ) {
   // Ui States handled by viewModel
   val uiState by viewModel.uiState.collectAsState()
-  LaunchedEffect(key1 = uiState.eventList) { viewModel.getEvents() }
-  val context = LocalContext.current
+
+  LaunchedEffect(Unit) { viewModel.getEvents() }
```

Comment:
nit: maybe it'a better to initialize this in the viewModel using `init{}` for separation of concerns?

Reply:
LaunchedEffect with Unit as the key ensures that getEvents() is called once when the composable enters the composition and not again unless the composable is recomposed from scratch. 
Triggering data loading in the ViewModel's init{} block can lead to premature requests, misaligned lifecycles, and unnecessary resource consumption.